### PR TITLE
Update ports when upgrade to newer CR version

### PR DESCRIPTION
**Pull request by blez**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

File: pkg/controller/pxc/controller.go [hunk 1]
```diff
@@ -288,6 +288,22 @@ func (r *ReconcilePerconaXtraDBCluster) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
+	for _, pxcService := range []corev1.Service{*pxc.NewServicePXC(o), *pxc.NewServicePXCUnready(o)} {
```
Comment (qJkee) [inline pkg/controller/pxc/controller.go:291]: should we really query this objects?
We need only Name and namespace from them, which is simply
Namespace: o.Namespace
Name: o.Name+"-"+appName

Comment (qJkee) [inline pkg/controller/pxc/controller.go:291]: use `[]*corev1.Service` and we will not need to add `*` to each array memeber

**Review by qJkee — APPROVED** [review]

**Review by tplavcic — DISMISSED** [review]

**Review by dAdAbird — DISMISSED** [review]

**Review by hors — APPROVED** [review]

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| haproxy | passed |
| recreate | passed |
| big-data | passed |
| storage | passed |
| self-healing | passed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| scaling-proxysql | passed |
| self-healing-advanced | passed |
| monitoring | passed |
| upgrade | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| upgrade-consistency | failed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| security-context | passed |
| users | passed |
| scheduled-backup | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| haproxy | passed |
| scaling | passed |
| recreate | passed |
| storage | passed |
| self-healing | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| self-healing-advanced | passed |
| upgrade-consistency | failed |
| monitoring | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| security-context | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:PR-631-a34dd3c7

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| storage | passed |
| recreate | passed |
| haproxy | passed |
| big-data | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| self-healing | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| self-healing-advanced | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |
| users | passed |
| scheduled-backup | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| demand-backup-encrypted-with-tls | passed |

**Pull request merged**